### PR TITLE
pull-request-guidelines.rst: explicitly mention backdoor and debug policy

### PR DESCRIPTION
**Pull request by pohly**
Ostro OS needs to be secure by default and code review needs to ensure
that.

Signed-off-by: Patrick Ohly <patrick.ohly@intel.com>

### CODE REVIEW

File: doc/howtos/pull-request-guidelines.rst [hunk 1]
```diff
@@ -168,3 +168,7 @@ Dos and Don'ts
   happens only for the full set of commits that are getting merged, so
   when bisecting history, the merge commits are good candidates for
   testing because they are more likely to build correctly.
+
+* Never add backdoors (= unexpected access to the system), debug features,
```
Comment (dbkinder) [inline doc/howtos/pull-request-guidelines.rst]: Never add backdoors or debug features for unexpected access to the system to the default configuration.  Such capabilities may be enabled only in explicit debug configurations of the OS images.

Comment (dbkinder) [inline doc/howtos/pull-request-guidelines.rst]: Do you have a convention for identifying and enabling these kinds of debug features?

Comment (pohly) [inline doc/howtos/pull-request-guidelines.rst]: There's no specific convention. Typically each optional feature comes with its own condition check.

Comment (pohly) [inline doc/howtos/pull-request-guidelines.rst]: The proposed rewording (IMHO) incorrectly talks about "debug features for unexpected access" - "debug features" are never allowed in the default configuration, regardless whether they enabled unexpected access or not.

How about:

* Never add backdoors (i.e., something that enables unexpected and unauthorized access to the system
  or some of its features) or debug features to the default configuration. Such capabilities may be enabled
  only in explicit debug configurations of the OS images.

Comment (dbkinder) [inline doc/howtos/pull-request-guidelines.rst]: +1

**Comment by pohly** [comment]
@kad okay to merge?

**Comment by kad** [comment]
For me, yes. but @dbkinder  is more relevant reviewer for such documentation changes

**Comment by dbkinder** [comment]
I did review the words.  If there's no problem talking about such issues then this PR is good to go.

**Pull request merged**